src/common/AndroidUtils.ts

```diff
@@ -663,21 +662,23 @@ export class AndroidSDKUtils {
         });
     }
 
-    private static logger: Logger = new Logger(
-        'force:lightning:mobile:android'
-    );
+    private static logger: Logger = new Logger(LOGGER_NAME);
     private static DEFAULT_ADB_CONSOLE_PORT = 5554;
     private static packageCache: Map<string, AndroidPackage> = new Map();
+    private static toolsBinLocation: string;
 
     private static getToolsBin(): string {
-        let toolsLocation = AndroidSDKUtils.ANDROID_TOOLS_BIN;
-        if (
-            !fs.existsSync(toolsLocation) &&
-            fs.existsSync(AndroidSDKUtils.ANDROID_CMD_LINE_TOOLS_BIN)
-        ) {
-            toolsLocation = AndroidSDKUtils.ANDROID_CMD_LINE_TOOLS_BIN;
+        if (this.toolsBinLocation === undefined) {
```

Comment:
Just changing this to be cached. There's actually a fair amount of performance improvement we could do on the Android side by not so generously running commands.